controller/api/destination/watcher/ip_watcher.go

```diff
@@ -93,6 +95,18 @@ func NewIPWatcher(k8sAPI *k8s.API, endpoints *EndpointsWatcher, log *logging.Ent
 		},
 	})
 
+	k8sAPI.Pod().Informer().AddIndexers(cache.Indexers{hostIPIndex: func(obj interface{}) ([]string, error) {
+		if pod, ok := obj.(*corev1.Pod); ok {
+			// If the pod does not have a host IP it should not be added to
+			// the host IP indexer.
+			if pod.Status.HostIP != "" {
+				return []string{pod.Status.HostIP}, nil
+			}
+			return []string{}, nil
+		}
+		return []string{""}, fmt.Errorf("object is not a pod")
```

Comment:
I think we can return `nil` here? In the error case, we have no reason to return a valid value

```suggestion
		return nil, fmt.Errorf("object is not a pod")
```